clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/ClusterStatsAggregatorTest.java

```diff
@@ -1,217 +1,142 @@
 // Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespa.clustercontroller.core;
 
+import com.google.common.collect.Sets;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.runners.MockitoJUnitRunner;
 
 import java.util.*;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.Matchers.any;
-import static org.mockito.Mockito.*;
 
 /**
  * @author hakonhall
  * @since 5.34
  */
-@RunWith(MockitoJUnitRunner.class)
 public class ClusterStatsAggregatorTest {
 
-    final Set<Integer> distributors = new HashSet<>();
-    final Set<Integer> storageNodes = new HashSet<>();
-    final Map<Integer, String> hostnames = new HashMap<>();
-    final MetricUpdater updater = mock(MetricUpdater.class);
-    ContentClusterStats clusterStats;
+    private static class StatsBuilder {
+        private final Map<Integer, Map<String, ContentNodeStats.BucketSpaceStats> > stats = new HashMap<>();
 
-    private void addDistributors(Integer... indices) {
-        for (Integer i : indices) {
-            distributors.add(i);
+        public StatsBuilder add(int nodeIndex, String bucketSpace, long bucketsTotal, long bucketsPending) {
+            return add(nodeIndex, bucketSpace, new ContentNodeStats.BucketSpaceStats(bucketsTotal, bucketsPending));
         }
-    }
-
-    private static class StorageNodeSpec {
-        public StorageNodeSpec(Integer index, String hostname) {
-            this.index = index;
-            this.hostname = hostname;
+        public StatsBuilder add(int nodeIndex, String bucketSpace) {
+            return add(nodeIndex, bucketSpace, new ContentNodeStats.BucketSpaceStats());
+        }
+        public StatsBuilder add(int nodeIndex, String bucketSpace, ContentNodeStats.BucketSpaceStats bucketSpaceStats) {
+            Map<String, ContentNodeStats.BucketSpaceStats> contentNodeStats = stats.get(nodeIndex);
+            if (contentNodeStats == null) {
+                contentNodeStats = new HashMap<>();
+                stats.put(nodeIndex, contentNodeStats);
+            }
+            contentNodeStats.put(bucketSpace, bucketSpaceStats);
+            return this;
+        }
+        public StatsBuilder add(int nodeIndex) {
+            stats.put(nodeIndex, new HashMap<>());
+            return this;
+        }
+        public ContentClusterStats build() {
+            Map<Integer, ContentNodeStats> nodeToStatsMap = new HashMap<>();
+            stats.forEach((nodeIndex, bucketSpaces) ->
+                    nodeToStatsMap.put(nodeIndex, new ContentNodeStats(nodeIndex, bucketSpaces)));
```

Comment:
Could also have used something like `stats.stream().collect(Collectors.toMap(e -> e.getKey(), e-> new ContentNodeStats(e.getKey(), e.getValue())));` to avoid explicit temporary map, but this is fine too (and has better named lambda variable names).